### PR TITLE
[Non-breaking changes] Move Message struct from types to core package

### PR DESCRIPTION
**Pull request by trinhdn2**
Reference:
- https://github.com/ethereum/go-ethereum/pull/25977

Checklist:
- [X] Encapsulate all the `Message` related structures & abstractions within a single package (core). Previously it was leaking into `core/types/Transaction.go`, most of whose users have no need for it
- [X] Fix some minor unit tests

### CODE REVIEW

File: core/headerchain.go [hunk 1]
```diff
@@ -66,9 +66,10 @@ type HeaderChain struct {
 }
 
 // NewHeaderChain creates a new HeaderChain structure.
-//  getValidator should return the parent's validator
-//  procInterrupt points to the parent's interrupt semaphore
-//  wg points to the parent's shutdown wait group
+//
+//	getValidator should return the parent's validator
```
Comment (imterryyy) [inline core/headerchain.go]:
```suggestion
// getValidator should return the parent's validator
```

File: core/headerchain.go [hunk 2]
```diff
@@ -66,9 +66,10 @@ type HeaderChain struct {
...
+//	procInterrupt points to the parent's interrupt semaphore
```
Comment (imterryyy) [inline core/headerchain.go]:
```suggestion
// procInterrupt points to the parent's interrupt semaphore
```

File: core/headerchain.go [hunk 3]
```diff
@@ -66,9 +66,10 @@ type HeaderChain struct {
...
+//	wg points to the parent's shutdown wait group
```
Comment (imterryyy) [inline core/headerchain.go]:
```suggestion
// wg points to the parent's shutdown wait group
```

File: core/state_processor.go [hunk 4]
```diff
@@ -517,7 +515,6 @@ func InitSignerInTransactions(config *params.ChainConfig, header *types.Header,
 		go func(from int, to int) {
 			for j := from; j < to; j++ {
 				types.CacheSigner(signer, txs[j])
```
Comment (imterryyy) [inline core/state_processor.go]: Keep cache hash for Tx
```suggestion
				types.CacheSigner(signer, txs[j])
				txs[j].CacheHash()
```

Comment (trinhdn2) [reply core/state_processor.go]: Re-added the `CacheHash`, PTAL

**Review by imterryyy — APPROVED** [review]

**Review by imterryyy — COMMENTED** [review]
LGTM

**Review by imterryyy — APPROVED** [review]

**Pull request closed without merging**